DataFormats/PatCandidates/interface/JetCorrFactors.h

```diff
@@ -35,6 +35,8 @@
 namespace pat {
 
   class JetCorrFactors {
+    friend class Jet;
```

Comment:
why is this needed?

Reply:
otherwise JecCorrFactors don't allow addition of new levels after creation

Reply:
it would be still better not to expose data members outside of the class.
Is there a significant problem to add a relevant modifier method?